modules/atomic-widgets/base/atomic-widget-base.php

```diff
@@ -182,17 +182,16 @@ private function transform_setting( $setting ) {
 					? join( ' ', $setting['value'] )
 					: '';
 
-			case 'image-attachment':
-				$attachment_url = wp_get_attachment_url( $setting['value']['id'] );
+			case 'image':
+				if ( isset( $setting['value']['url'] ) ) {
+					$attachment_url = $setting['value']['url'];
+				}
 
-				return [
-					'url' => $attachment_url ?? Utils::get_placeholder_image_src(),
-				];
+				if ( isset( $setting['value']['attachmentId'] ) ) {
+					$attachment_url = wp_get_attachment_image_url( $setting['value']['attachmentId'] ) ?? null;
+				}
 
-			case 'image-url':
-				return [
-					'url' => $setting['value']['url'] ?? Utils::get_placeholder_image_src(),
-				];
+				return $attachment_url ?? Utils::get_placeholder_image_src();
```

Comment:
shouldn't it be something like this?
(because you need to have either id or url)

```PHP
$url = '';

if ( isset( $value['id'] ) ) {
	$url = get_by_id( $value['id'] );
} else if ( isset( $value['url' ) ) {
	$url = $value['url'];
}

return $url ?? get_placeholder();
```

Reply:
Done